lib/gitlab/ci/templates/pipeline/AndroidTemplate.yml

```diff
@@ -73,6 +75,48 @@ deployDebugApk:
         https://slack.com/api/files.upload
   cache: {}
 
+testRelease:
+  stage: test
+  only:
+    - tags
+  script:
+    - ./gradlew -Pci --console=plain testRelease
+  artifacts:
+    when: always
+    reports:
+      junit:
+        - "**/TEST-*.xml"
+
+assembleReleaseApk:
+  stage: assembleRelease
+  only:
+    - tags
+  script:
+    - echo ${KEYSTORE_FILE} | base64 -d > my.keystore
+    - ./gradlew assembleRelease
+      -Pandroid.injected.signing.store.file=$(pwd)/my.keystore
+      -Pandroid.injected.signing.store.password=$KEYSTORE_PASSWORD
+      -Pandroid.injected.signing.key.alias=$KEY_ALIAS
+      -Pandroid.injected.signing.key.password=$KEY_PASSWORD
+  artifacts:
+    paths:
+      - app/build/outputs/apk/release
+
+deployReleaseApk:
+  stage: deployRelease
+  only:
+    - tags
+  script:
```

Comment:
So this job is more configurable for other future projects, maybe change it to something like the below?  Also, if adding the below please also add `DEPLOY_DEBUG_APK_SLACK_MESSAGE`, `DEPLOY_DEBUG_APK_PATH`, and `DEPLOY_DEBUG_APK_NAME` to the `Gradle Android Pipeline` documentation at https://github.com/chesapeaketechnology/gitlab-templates/blob/release/1.3/README.md#customization-1. 
```
  script:
    - if [ -z "${APK_SLACK_CHANNEL_ACCESS_TOKEN}" ]; then echo "APK_SLACK_CHANNEL_ACCESS_TOKEN is unset so not pushing message to Slack"; else echo "APK_SLACK_CHANNEL_ACCESS_TOKEN is set to '$APK_SLACK_CHANNEL_ACCESS_TOKEN'"; fi
    - if [ -z "${APK_SLACK_CHANNEL_ID}" ]; then echo "APK_SLACK_CHANNEL_ID is unset so not pushing message to Slack"; else echo "APK_SLACK_CHANNEL_ACCESS_TOKEN is set to '$APK_SLACK_CHANNEL_ID'"; fi
    - if [ -z "${DEPLOY_RELEASE_APK_SLACK_MESSAGE}" ]; then echo "$DEPLOY_RELEASE_APK_SLACK_MESSAGE is unset so will use empty message for Slack"; else echo "$DEPLOY_RELEASE_APK_SLACK_MESSAGE is set to '$DEPLOY_RELEASE_APK_SLACK_MESSAGE'"; fi
    - release_apk_file=$(find $DEPLOY_RELEASE_APK_PATH -name *$DEPLOY_RELEASE_APK_NAME*)
    - echo "found release apk file $release_apk_file"
    - |
      curl \
        -F token="${APK_SLACK_CHANNEL_ACCESS_TOKEN}" \
        -F channels="${APK_SLACK_CHANNEL_ID}" \
        -F initial_comment="$DEPLOY_RELEASE_APK_SLACK_MESSAGE" \
        -F "file=@$release_apk_file" \
        https://slack.com/api/files.upload
```